wagtail/tests/utils.py

```diff
@@ -51,3 +60,122 @@ def reset_warning_registry():
+        """
+        if self._testCanCreateAt(parent_model, child_model):
+            msg = self._formatMessage(msg, "Can create a %s.%s under a %s.%s" % (
+                child_model._meta.app_label, child_model._meta.model_name,
+                parent_model._meta.app_label, parent_model._meta.model_name))
+            raise self.failureException(msg)
+
+    def assertCanCreate(self, parent, child_model, data, msg=None):
+        """
+        Assert that a child of the given Page type can be created under the
+        parent, using the supplied POST data.
+
+        ``parent`` should be a Page instance, and ``child_model`` should be a
+        Page subclass. ``data`` should be a dict that will be POSTed at the
+        Wagtail admin Page creation method.
+        """
+        self.assertCanCreateAt(parent.specific_class, child_model)
+
+        if 'slug' not in data and 'title' in data:
+            data['slug'] = slugify(data['title'])
+        data['action-publish'] = 'action-publish'
+
+        add_url = reverse('wagtailadmin_pages:add', args=[
+            child_model._meta.app_label, child_model._meta.model_name, parent.pk])
+        response = self.client.post(add_url, data, follow=True)
+
+        if response.status_code != 200:
+            msg = self._formatMessage(msg, 'Creating a %s.%s returned a %d' % (
+                child_model._meta.app_label, child_model._meta.model_name, response.status_code))
+            raise self.failureException(msg)
+
+        if response.redirect_chain == []:
+            if 'form' not in response.context:
+                msg = self._formatMessage(msg, 'Creating a page failed unusually')
+                raise self.failureException(msg)
+            form = response.context['form']
+            if not form.errors:
+                msg = self._formatMessage(msg, 'Creating a page failed for an unknown reason')
+                raise self.failureException(msg)
+
+            errors = '\n'.join('  %s:\n    %s' % (field, '\n    '.join(errors))
+                               for field, errors in sorted(form.errors.items()))
+            msg = self._formatMessage(msg, 'Validation errors found when creating a %s.%s:\n%s' % (
+                child_model._meta.app_label, child_model._meta.model_name, errors))
+            raise self.failureException(msg)
+
+        if django.VERSION >= (1, 9):
+            explore_url = reverse('wagtailadmin_explore', args=[parent.pk])
+        else:
+            explore_url = 'http://testserver' + reverse('wagtailadmin_explore', args=[parent.pk])
+        if response.redirect_chain != [(explore_url, 302)]:
+            msg = self._formatMessage(msg, 'Creating a page %s.%s didnt redirect the user to the explorer, but to %s' % (
+                child_model._meta.app_label, child_model._meta.model_name,
+                response.redirect_chain))
+            raise self.failureException(msg)
+
+    def assertAllowedSubpageTypes(self, parent_model, child_models, msg=None):
```

Comment:
These were for two reasons: to keep the ugly ContentType conversions out of user code, and to abstract the actual mechanism Wagtail uses to check these rules. These tests are primarily for users who are not experts on the Wagtail internals, easily accessible methods like these are written with those users in mind.

`clean_subpage_models` is an internal Wagtail method that is not documented. I named this public-facing method something that would make sense to those not familiar with the Wagtail internals. But now that I read this code again, I was supposed to be testing `allowed_subpage_modes`, not `clean_subpage_models`! I'll fix that right up.